test_quality_of_service/CMakeLists.txt

```diff
@@ -44,6 +44,7 @@ if(BUILD_TESTING)
 
   ament_find_gtest()
   target_include_directories(${PROJECT_NAME}_support PUBLIC "${GTEST_INCLUDE_DIRS}")
+  target_link_libraries(${PROJECT_NAME}_support ${GTEST_LIBRARIES})
```

Comment:
So we ended up removing this in #514 because we were having ODR issues on Linux and Windows.  We'll need to make sure that that doesn't happen again.

But that said, I think a better way to go here is to actually get rid of the need for `gtest` in this support library at all.  Only `BaseQosRclcppTestFixture` in `qos_utilities.{cpp,hpp}` ends up using it.  So I think what we should probably do instead is to remove that particular class from that utility method, and just replicate it into the 4 places that need it (`test_deadline.cpp`, `test_lifespan.cpp`, `test_liveliness.cpp`, and `test_best_available.cpp`).  Then we can also remove the `target_include_directories` above, and that solution should definitely avoid ODR and also make this work for macOS.

That said, I actually don't see why we need this at all.  As far as I can tell, none of the source files that make up `${PROJECT_NAME}_support` actually depend on `gtest`, so they shouldn't link against it.  In point of fact, I would also remove the `target_include_directories` line above, since that shouldn't be needed either.

Reply:
You're right: qos_test_{node|publisher|subscriber} don't need it at all. Instead of copying BaseQosRclcppTestFixture to the four tests, in CMakeLists.txt I moved `qos_utilities.cpp` from `${PROJECT_NAME}_support` to the `add_custom_gtest` calls. This avoids the suggested code redundancy, but it needs to be compiled 4 times though.

Reply:
Yeah, good call.  This will do.

Can you please add a Signed-off-by line to your commit (`git commit --amend -s`)?  Once that is done, I'll approve and run CI on it.

Reply:
Of course. It's added and pushed.

Reply:
The ubuntu jammy amd64 test fails because of latest commit https://github.com/ros2/system_tests/commit/d58707976ef51d049288b9d5bf9da681fe51ced1. Shall I rebase to previous commit or is it fine?